adapter/go.mod

```diff
@@ -1,11 +1,10 @@
 module github.com/wso2/product-microgateway/adapter
 
-go 1.15
+go 1.18
```

Comment:
shall we update develper guide/prerequisites as well

Reply:
updated. Thanks